#! /usr/bin/env python

PACKAGE='controller_adaptiveclbf'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name       Type      Level Description                Default Min   Max
gen.add("enable",  bool_t,   0,    "Enable node operation.",  True)

gen.add("width", double_t, 0, "Figure 8 width (m)", 4.0, 0.0,100.0)
gen.add("height", double_t, 0, "Figure 8 height (m)", 4.0, 0.0,100.0)
gen.add("speed", double_t, 0, "Figure 8 speed (m/s)", 1.0, 0.01,100.0)
gen.add("freq", double_t, 0, "Oscillation frequency", 0.25, 0.001,100.0)
gen.add("x_offset", double_t, 0, "Figure 8 center x(m)", 0.0, -1000.0,1000.0)
gen.add("y_offset", double_t, 0, "Figure 8 center y(m)", 0.0, -1000.0,1000.0)

exit(gen.generate(PACKAGE, "controller_adaptiveclbf", "figureEight"))
